openproblems/tasks/dimensional_reduction/README.md

```diff
@@ -1,11 +1,16 @@
-### Dimensional reduction manifold preservation
----
+## Dimensional Reduction
 
-Dimensional reduction is one of the key challenges in single-cell data representation. Routine single-cell RNA sequencing (scRNA-seq) experiments measure cells in roughly 20,000-30,000 dimensions (i.e., features - mostly gene transcripts but also other functional elements encoded in mRNA such as lncRNAs.) Since its inception, scRNA-seq experiments have been growing in terms of the number of cells measured. Originally, cutting-edge SmartSeq experiments would yield <a href="#">a few hundred cells</a>, at best. Now, it is not uncommon to see experiments that yield over <a href="https://www.nature.com/articles/s41586-018-0590-4">100,000 cells</a> or even <a href="https://www.10xgenomics.com/blog/our-13-million-single-cell-dataset-is-ready-to-download">> 1 million cells</a>.
+Dimensional reduction is a key challenge in single-cell data representation [<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02128-7">1</a>].
 
-Each *feature* in a dataset functions as a single dimension. In the AnnData structure, these are stored in the `adata.var` component. Each cell analyzed is stored in `adata.obs`. The measured cells are thereby represented in the **cell x gene expression matrix**, `adata.X`.
+## API
 
-![AnnData Structure](images/core_anndata.png)
+Datasets should contain the following attributes:
 
-While each of the ~30,000 dimensions measured in each cell (not accounting for roughly 75-90% data dropout per cell, another issue entirely), likely contribute to some sort of data structure, the overall structure of the data is diluted due to the <a href = "https://en.wikipedia.org/wiki/Curse_of_dimensionality">*curse of dimensionality*</a>. In short, it's difficult to visualize the contribution of each individual gene in a way that makes sense to the human eye, i.e., two or three dimensions (at most). Thus, we need to find a way to <a href = "https://en.wikipedia.org/wiki/Dimensionality_reduction">*dimensionally reduce*</a> the data for visualization and interpretation.
+* `adata.obs["labels"]` (cell/observation labels)
+* `adata.var["labels"]` (gene name feature labels)
 
+**Methods** should assign dimensionally-reduced embedding coordinates to `adata.obsm['dimensional_reduction']` i.e., `adata.obsm['X_pca']` or `adata.obsm['X_tsne']`.
```

Comment:
I guess this should now say "assign [...] to adata.obsm['X_emb']" ?

Reply:
Also, we should check this exists in `checks.py:check_method` (see https://github.com/singlecellopenproblems/SingleCellOpenProblems/blob/master/openproblems/tasks/label_projection/checks.py for an example.)

Reply:
great catch - fixed and implemented. Might need tweaking but I think I got it right.